src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -706,39 +718,36 @@ def _convert_tokens(  # noqa: PLR0913
                 )
             token_infos.append(token_info)
 
-    async def _validate_prompt_and_tokenize(
+    async def _validate_prompt_and_tokenize(  # noqa: PLR0913
         self,
         sampling_params: SamplingParams,
         truncate_input_tokens: int | None,
         prompt: str,
+        tokenizer: PreTrainedTokenizer,
         context: ServicerContext,
     ) -> tuple[list[int], bool]:
         assert self.config is not None
 
         max_model_len = self.config.max_model_len
-        # tokenize_kwargs = {"truncation": True,
-        #                    "max_length": truncate_input_tokens} \
-        #     if truncate_input_tokens is not None else {
-        #       "truncation": True, "max_length": max_model_len + 1}
-        tokenize_kwargs: dict[str, Any] = {}
-
-        input_ids = await self.tokenizer_group.encode_async(
-            prompt,
-            **tokenize_kwargs,
-        )
 
-        # TODO this is temporary until truncation option is added
-        # to the TokenizerGroup encode methods
-        if truncate_input_tokens and truncate_input_tokens < len(input_ids):
-            input_ids = input_ids[-truncate_input_tokens:]
-            if not sampling_params.skip_special_tokens:
```

Comment:
This was actually a bug, it shouldn't have been checking this param here (it applies to output tokens not input tokens).